src/nvim/screen.c

```diff
@@ -3949,40 +3971,84 @@ win_line (
             && (int)wp->w_virtcol <
             wp->w_width * (row - startrow + 1) + v
             && lnum != wp->w_cursor.lnum)
-           || draw_color_col)
-          && !wp->w_p_rl
-          ) {
+           || draw_color_col || do_eoltext)
+          && !wp->w_p_rl) {
         int rightmost_vcol = 0;
         int i;
+        size_t eol_pos = 0;
+        LineState s = LINE_STATE((char_u *)"");
+        int eol_attr = 0;
+
+        VirtText virt_text = do_eoltext ? bufhl_info.line->virt_text
+                                        : (VirtText)KV_INITIAL_VALUE;
+        bool delay_text = lcs_eol <= 0;
```

Comment:
when is `lcs_eol <= 0` ? Would it make sense to rename `delay_text` to `delay_virttext_for_lcs` ?